tests/testthat/test-harness.R

```diff
@@ -0,0 +1,93 @@
+test_that("test harness and kelp advertise same interface", {
+  harness <- kelp_harness$new(seaweed_master_url)
+  real <- kelp$new(seaweed_master_url)
+
+  harness_funcs <- ls(harness)
+  kelp_funcs <- ls(real)
+  expect_equal(harness_funcs, kelp_funcs)
+  for (name in harness_funcs) {
+    harness_member <- harness[[name]]
+    kelp_member <- real[[name]]
+    if (is.function(harness_member)) {
+      expect_equal(args(harness_member), args(kelp_member))
+    } else {
+      expect_equal(harness_member, kelp_member)
+    }
+  }
+})
+
+test_that("test harness: can upload, read and delete a file", {
+  test_seaweed_available()
+  fs <- kelp_harness$new(seaweed_master_url)
+
+  ## File can be written
+  t <- tempfile()
+  writeLines("test file", t)
+  res <- fs$upload_file(t)
+  expect_match(res, "\\d+,[A-Za-z0-9]{10}")
```

Comment:
haha, so similar in our regexs. Consider putting `^` at the front, `$` at the back